src/mono/mono/mini/aot-compiler.c

```diff
@@ -14830,6 +14831,8 @@ aot_assembly (MonoAssembly *ass, guint32 jit_opts, MonoAotOptions *aot_options)
 
 	if (acfg->aot_opts.compiled_methods_outfile && acfg->dedup_phase != DEDUP_COLLECT) {
 		acfg->compiled_methods_outfile = fopen (acfg->aot_opts.compiled_methods_outfile, "w+");
+		fprintf(acfg->compiled_methods_outfile, "%s\n", ass->aname.name);
```

Comment:
Might be better to include `ass->image->filename` instead, so filename construction/checks around https://github.com/dotnet/runtime/pull/86722/files#diff-b005e8c8e4597bf10080a34f426823c8bb6da196e462c3119a66ba3b0ee03e93R142 would be simpler.